### PR TITLE
[libsercomm] Adding new port

### LINKED ISSUE #21003: [New Port Request] libsercomm
Library name: libsercomm

Library description: A Multiplatform Serial Communications Library

Source repository URL: https://github.com/ingeniamc/sercomm

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

* Be build for at least windows x64, linux x64 and mac x64

----------------------------

If needed I could fork it. The maintainer of the original repo has not been active in four years.
If there are any other library out there or already included in vcpkg (I haven't seen any) this is not needed to be ported.

I am planing to look into this when I have time, but anyone please feel free to give it a try if you want to!

### PR DESCRIPTION
**Pull request by misirlou-tg**
Adds a new port, libsercomm

- #### What does your PR fix?  
  Fixes #21003 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libsercomm/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        errdesc WITH_ERRDESC
+        devmon  WITH_DEVMON
+        pic     WITH_PIC
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/libsercomm/portfile.cmake:19]:
```suggestion
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libsercomm/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    SHA512 f1581f2dfa262ffb1b3aec5a1e6d32493c322c94541fbacc98efff23b3b42b14c9abdcfb063a78b7c54fb1f9d8dbf59d8064099601de2175af6c6d830708324c
+    HEAD_REF master
+)
+
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        errdesc WITH_ERRDESC
+        devmon  WITH_DEVMON
+        pic     WITH_PIC
```
Comment (dg0yt) [inline ports/libsercomm/portfile.cmake:14]: Does it really make sense to expose this "feature"? Most toolchains use the `-fpic` compiler flag.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static
- x64-linux

**Comment by NangiDev** [comment]
Thank you so much!
Great work!

What would be needed to do to get the osX triplet working?
Also other triplets if that's possible.

**Comment by GongT** [comment]
I can't use this package.

It install `libsercomm/sercommConfig.cmake`, but `find_package` want ether `libsercomm/libsercommConfig.cmake` or `sercomm/sercommConfig.cmake`

```
[cmake] CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:800 (_find_package):
[cmake]   Could not find a package configuration file provided by "sercomm" with any
[cmake]   of the following names:
[cmake] 
[cmake]     sercommConfig.cmake
[cmake]     sercomm-config.cmake
[cmake] 
[cmake]   Add the installation prefix of "sercomm" to CMAKE_PREFIX_PATH or set
[cmake]   "sercomm_DIR" to a directory containing one of the above files.  If
[cmake]   "sercomm" provides a separate development package or SDK, be sure it has
[cmake]   been installed.
[cmake] Call Stack (most recent call first):
[cmake]   CMakeLists.txt:49 (find_package)
```

**Comment by JonLiu1993** [comment]
@GongT ,Please post a new issue to describle your question, and we will track your question in your issue

**Pull request merged**